ci/prow/unit-tests

```diff
@@ -3,3 +3,4 @@
 set -euxo pipefail
 
 make unit-test
+make manager
```

Comment:
Why is this required?

If it is only to test the the code is compiled correctly, then `unit-test` will fail (because `vet` will fail)

Reply:
Shouldn't it be part of the `lint` job in that case?

Reply:
`lint` runs in a golangci-lint image, not sure it can build our code and with which go version.
Unit tests do not need to build the manager binary to run, so I wouldn't add it as a dependency.
Maybe we should have a separate `build` job then?

Reply:
> Maybe we should have a separate build job then?

I think so if you don't mind.